readthedocs/core/middleware.py

```diff
@@ -79,10 +79,10 @@ def process_request(self, request):
                 'localhost' not in host and
                 'testserver' not in host):
             request.cname = True
-            domains = Domain.objects.filter(domain=host)
-            if domains.count():
+            domains = Domain.objects.filter(domain=full_host)
```

Comment:
I did a quick look in prod to see if anybody had a domain with a `:` in it as this change would affect what they're doing. There is one but it appears to be a buggy project.